docs/guides/Screen-Navigation-Prop.md

```diff
@@ -1,16 +1,28 @@
 
 # Screen Navigation Prop
 
-Each screen in your app will recieve a navigation prop, which contains the following:
+Each *screen* in your app will receive a navigation prop which contain the following:
+* `navigate` - (helper) link to other screens
+* `state` - screen's current state/routes
+* `setParams` - (helper) make changes to route's params
+* `goBack` - (helper) close active screen and move back
+* `dispatch` - send an action to router
 
+*NOTE:* The `navigation` prop is passed down to every navigation-aware component including navigators. The big exception is that a navigator's `navigation` prop may not have the helper functions (`navigate`, `goBack`, etc); it may only have `state` and `dispatch`. In order to `navigate` using the navigator's `navigation` prop, you will have to `dispatch` using an [action creator](navigation-actions).
+
+*Notes regarding hooking things up with Redux*
+
+> People don't always hook things up to redux correctly, because they mis-understand the navigator's top-level API, where the navigation prop is optional. The navigator will maintain its own state if it doesn't get a navigation prop, but this is not a feature you generally want to use when hooking your app up with redux. For navigators that are nested inside of your main navigator, you always want to pass the screen's navigation prop down. This allows your top-level navigator to communicate and provide state for all the children navigators. Only your top-level router needs to be integrated with redux, because all the other routers are inside it.
```

Comment:
hey @ericvicenti I just left this part unedited because I didn't quite understand what was going on with the redux integrations. According to the [redux integration guide](https://reactnavigation.org/docs/guides/redux) I read it as: if you pass in a navigation prop to a `Navigator`:
1. it effectively replaces the router -> meaning you need to implement `getStateForProps`, `getStateForAction` not that they will actually be called since `react-navigation`'s actions aren't being used
2. All the built in actions like `Back`, `Reset` need to be re-implemented or copied inside your own app when creating your redux store/connecting to app

According to your notes when hooking up with Redux, you can hook up just the top level Navigator to Redux and still have the child Navigators' Routers with `react-navigation`'s router api..?

There must be something I am missing, could you please point me in the right direction?